### PR TITLE
mod conversion: add converter and unit

### LINKED ISSUE #32: add mod for currency conversion
Struct Pricer or Rater with currency conversion method.
Should be in view::Context and use in views

### PR DESCRIPTION
**Pull request by edouardparis**
A converter converts a given amount in satoshis
to another amount in a chosen unit.
This struct can be enhanced with currency
conversion.

Converter is shared to each view through the Context,
because it is only a display feature and does not have
core revault logic.

close #32

### CODE REVIEW

File: src/ui/view/mod.rs [hunk 1]
```diff
@@ -30,12 +34,30 @@ pub struct Context {
...
+    pub fn default() -> Context {
+        Context {
+            converter: Converter::new(Network::Bitcoin),
+            network: Network::Bitcoin,
+            network_up: false,
+            role: Role::Manager,
+            menu: Menu::Home,
+            role_edit: false,
+        }
+    }
```
Comment (darosior) [inline src/ui/view/mod.rs:46]: nit: Implementing `Default` seems more rustic?

**Review by darosior — APPROVED** [review]
utACK 7fd7d5df322a14c9d75d81334d01db10aa44b351

Nice cleanup. Just have a question and nit, ff to move forward without modifications though.

File: src/conversion.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        };
+        Conversion { amount, unit }
+    }
+}
+
+/// Conversion is the result of the Converter job.
+pub struct Conversion {
+    pub amount: f64,
+    pub unit: Unit,
+}
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply src/conversion.rs:32]: wow, didnt know about it thanks !

File: src/conversion.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        let unit = match bitcoin_network {
+            Network::Testnet => Unit::TestnetBitcoin,
+            Network::Bitcoin => Unit::Bitcoin,
+            Network::Regtest => Unit::RegtestBitcoin,
+        };
+        Self { unit }
+    }
+
+    /// converts amount in satoshis to BTC float.
+    pub fn converts(&self, amount: &u64) -> f64 {
```
Comment (darosior) [inline src/conversion.rs:20]: What's the point of taking a reference to an `u64` ?  Even more if you copy it on the next line..

**Review by darosior — APPROVED** [review]
utACK 3fea012f6c67eb9bcc64a8b3b3ce3cf6d4e3a4da

My comment imo should be fixed but it's not critical at all.

**Review by danielabrozzoni — APPROVED** [review]

File: src/ui/state/manager.rs [hunk 4]
```diff
@@ -321,9 +321,10 @@ impl ManagerSendOutput {
         if self.amount.is_empty() {
             return Ok(0);
         }
-        let a = f64::from_str(&self.amount)
+
+        let amount = bitcoin::Amount::from_str_in(&self.amount, bitcoin::Denomination::Bitcoin)
             .map_err(|_| Error::UnexpectedError("cannot parse output amount".to_string()))?;
-        Ok((a * 100000000_f64) as u64)
+        Ok(amount.as_sat())
```
Comment (darosior) [inline src/ui/state/manager.rs:327]: So much cleaner :D

**Review by darosior — APPROVED** [review]
utACK bcb1b433dcbcf3ebc5d5ab341b7856831e7640f5

File: src/ui/view/home.rs [hunk 5]
```diff
@@ -141,35 +146,31 @@ fn unsecured_fund_view<'a>(
...
-                    Container::new(text::bold(text::simple(&format!(
-                        "{}",
-                        balance.0 as f64 / 100000000_f64
-                    ))))
-                    .width(Length::Shrink),
+                    Container::new(text::bold(text::simple(&format!("{}", active_balance))))
+                        .width(Length::Shrink),
                 )
-                .push(text::simple(" BTC")),
+                .push(text::simple(&format!(" {}", ctx.converter.unit))),
```
Comment (danielabrozzoni) [inline src/ui/view/home.rs:162]: nit: `format!` is not necessary, you can use `to_string`:
```
.push(text::simple(&ctx.converter.unit.to_string())),
```

Comment (danielabrozzoni) [reply src/ui/view/home.rs:162]: (If you choose to use `to_string` instead, please update it in the other parts of the code where there is `format!("{}", something)`, so we remain consistent; feel free to discard this suggestion though :))

**Comment by danielabrozzoni** [comment]
ACK bcb1b433dcbcf3ebc5d5ab341b7856831e7640f5

File: src/ui/view/home.rs (hunk 5 above)
Comment (edouardparis) [reply src/ui/view/home.rs:162]: Thank you I will create an issue

**Pull request merged**